topic0keyPoint3/src/topic0keyPoint3/HospitalFactory.java

```diff
@@ -0,0 +1,17 @@
+package topic0keyPoint3;
+
+public class HospitalFactory {
+
+	public Hospital getAhospital(String hospitalType) {
+
+		if (hospitalType.equals("psychiatricHospital")) {
```

Comment:
Two things here:
if psychiatricHospital is null, could throw a null pointer exception. So the better way is invert the args
```
if ("psychiatricHospital".equalsIgnoreCase(hospitalType) {

}
```

Also see the method **equalsIgnoreCase()**, that ignore the cases...